Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIr.kt

```diff
@@ -0,0 +1,460 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+package org.jetbrains.kotlin.native.interop.gen
+
+import org.jetbrains.kotlin.native.interop.indexer.*
+
+interface StubIrElement {
+    fun accept(visitor: StubIrVisitor)
+
+    fun <T, R> accept(visitorAux: StubIrVisitorAux<T, R>, data: T): R
+}
+
+interface StubContainer : StubIrElement {
+    val meta: StubContainerMeta
+    val classes: List<ClassStub>
+    val functions: List<FunctionalStub>
+    val properties: List<PropertyStub>
+    val typealiases: List<TypealiasStub>
+    val simpleContainers: List<SimpleStubContainer>
+}
+
+/**
+ * Meta information about [StubContainer].
+ * For example, can be used for comments in textual representation.
+ */
+class StubContainerMeta(
+        val textAtStart: String = "",
+        val textAtEnd: String = ""
+)
+
+/**
+ * Trivial container of IR elements that can be used
+ * for module or ObjC category representation.
+ */
+class SimpleStubContainer(
+        override val meta: StubContainerMeta = StubContainerMeta(),
+        override val classes: List<ClassStub> = emptyList(),
+        override val functions: List<FunctionalStub> = emptyList(),
+        override val properties: List<PropertyStub> = emptyList(),
+        override val typealiases: List<TypealiasStub> = emptyList(),
+        override val simpleContainers: List<SimpleStubContainer> = emptyList()
+) : StubContainer {
+    override fun accept(visitor: StubIrVisitor) {
+        visitor.visitSimpleStubContainer(this)
+    }
+
+    override fun <T, R> accept(visitorAux: StubIrVisitorAux<T, R>, data: T): R {
+        return visitorAux.visitSimpleStubContainer(this, data)
+    }
+}
+
+val StubContainer.children: List<StubIrElement>
+    get() {
+        val classes: List<StubIrElement> = classes
+        val funcs: List<StubIrElement> = functions
```

Comment:
What are these variables required for?

Reply:
Because inferred type of concatenation was `List<Any>`. Made it simpler.